components/camera/transformpipeline/classifier.go

```diff
@@ -0,0 +1,105 @@
+	"context"
+	"fmt"
+	"image"
+
+	"github.com/edaniels/gostream"
+	"go.opencensus.io/trace"
+	"go.viam.com/rdk/components/camera"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/rimage/transform"
+	"go.viam.com/rdk/robot"
+	"go.viam.com/rdk/services/vision"
+	rdkutils "go.viam.com/rdk/utils"
+	"go.viam.com/rdk/vision/classification"
+)
+
+// classifierAttrs is the attribute struct for classifiers.
+type classifierAttrs struct {
+	ClassifierName      string  `json:"classifier_name"`
+	ConfidenceThreshold float64 `json:"confidence_threshold"`
+}
+
+// classifierSource takes an image from the camera, and overlays a label from the classifier.
+type classifierSource struct {
+	stream         gostream.VideoStream
+	classifierName string
+	confFilter     classification.Postprocessor
+	r              robot.Robot
+}
+
+func newClassificationsTransform(
+	ctx context.Context,
+	source gostream.VideoSource, r robot.Robot, am config.AttributeMap,
+) (gostream.VideoSource, camera.ImageType, error) {
+	conf, err := config.TransformAttributeMapToStruct(&(classifierAttrs{}), am)
+	if err != nil {
+		return nil, camera.UnspecifiedStream, err
+	}
+	attrs, ok := conf.(*classifierAttrs)
+	if !ok {
+		return nil, camera.UnspecifiedStream, rdkutils.NewUnexpectedTypeError(attrs, conf)
+	}
+
+	props, err := propsFromVideoSource(ctx, source)
+	if err != nil {
+		return nil, camera.UnspecifiedStream, err
+	}
+	var cameraModel transform.PinholeCameraModel
+	cameraModel.PinholeCameraIntrinsics = props.IntrinsicParams
+
+	if props.DistortionParams != nil {
+		cameraModel.Distortion = props.DistortionParams
+	}
+	confFilter := classification.NewScoreFilter(attrs.ConfidenceThreshold)
+	classifier := &classifierSource{
+		gostream.NewEmbeddedVideoStream(source),
+		attrs.ClassifierName,
+		confFilter,
+		r,
+	}
+	cam, err := camera.NewFromReader(ctx, classifier, &cameraModel, camera.ColorStream)
+	return cam, camera.ColorStream, err
+}
+
+// Read returns the image overlaid with at most one label from the classification. It overlays the
+// highest-confidence label along with the confidence score, as long as the score is above the
+// confidence threshold.
+func (cs *classifierSource) Read(ctx context.Context) (image.Image, func(), error) {
+	ctx, span := trace.StartSpan(ctx, "camera::transformpipeline::classifier::Read")
+	defer span.End()
+
+	srv, err := vision.FirstFromRobot(cs.r)
+	if err != nil {
+		return nil, nil, fmt.Errorf("source_classifier can't find vision service: %w", err)
+	}
+	// get image from source camera
+	img, release, err := cs.stream.Next(ctx)
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not get next source image: %w", err)
+	}
+	classifications, err := srv.Classifications(ctx, img, cs.classifierName, 1, map[string]interface{}{})
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not get classifications: %w", err)
+	}
+	// overlay label on the source image
+	classifications = cs.confFilter(classifications)
+	if len(classifications) > 1 {
```

Comment:
Good thinking, I made this configurable and included an image in the PR description showing multiple classifications.

The only remaining question here is whether we should cap the number of max classifications, or add logic to the `Overlay` method to exit once we've reached the outer bounds of the image. I tried making the max classification value 100 and the confidence score 0 (so all 100 actually show up on the image), and it just overlaid as many as it could fit. I guess the only concern here is that we'd still be looping over all of the classifications so this cause performance issues if the user made `max_classifications` some very large value.

Reply:
If it's easy enough to add logic to exit the Overlay method once it's reached the outer bounds of the image, that'd be ideal but otherwise I think fine to keep as-is.